cg/cli/upload/observations.py

```diff
@@ -1,105 +1,123 @@
+
+    click.echo(click.style("----------------- OBSERVATIONS -----------------"))
+
+    case: models.Family = get_observations_case(context, case_id)
+    observations_api: UploadObservationsAPI = get_observations_api(context, case)
+
+    if dry_run:
+        LOG.info(f"Dry run. Would upload observations for {case.internal_id}.")
+        return
+
+    try:
+        observations_api.process(case.analyses[0])
+    except (FileNotFoundError, DuplicateRecordError, DuplicateSampleError) as error:
+        LOG.error(f"Cancelling observations upload for case {case.internal_id}, {error.message}")
+        return
+
+    LOG.info(f"Observations uploaded for case: {case.internal_id}")
 
-    click.echo(click.style("----------------- OBSERVATIONS ----------------"))
-    status_db: Store = context.status_db
-    housekeeper_api: HousekeeperAPI = context.housekeeper_api
+
+@click.command("available-observations")
+@click.option(
+    "--pipeline",
+    type=click.Choice(LOQUSDB_SUPPORTED_PIPELINES),
+    help="Limit observations upload to a specific pipeline",
+)
+@click.option("--dry-run", is_flag=True, help="Only print the cases that would be processed")
+@click.pass_context
+def available_observations(context: click.Context, pipeline: Optional[Pipeline], dry_run: bool):
+    """Uploads the available observations to Loqusdb."""
+
+    click.echo(click.style("----------------- AVAILABLE OBSERVATIONS -----------------"))
+
+    status_db: Store = context.obj.status_db
+    cases_to_upload: Query = status_db.observations_to_upload(pipeline=pipeline)
+    if not cases_to_upload:
+        LOG.error(
+            f"There are no available cases to upload to Loqusdb for {pipeline} ({datetime.now()})"
+        )
+        return
+
+    for case in cases_to_upload:
+        context.invoke(observations, case=case.internal_id, dry_run=dry_run)
+
+
+def get_observations_case(context: CGConfig, case_id: str) -> models.Family:
+    """Return a valid Loqusdb case to upload given a case ID."""
+
+    case: models.Family = context.status_db.family(case_id)
+    if not case or case.data_analysis not in LOQUSDB_SUPPORTED_PIPELINES:
+        LOG.error("Invalid case ID. Retrieving available cases for Loqusdb upload.")
+        cases_to_upload: Query = context.status_db.observations_to_upload()
+        if not cases_to_upload:
+            LOG.info("There are no valid cases to upload to Loqusdb")
+        else:
+            LOG.info("Provide one of the following case IDs: ")
+            for case in cases_to_upload:
+                LOG.info(f"{case.internal_id} ({case.data_analysis})")
+        raise click.Abort()
+    elif not case.customer.loqus_upload:
+        LOG.error(
+            f"Customer {case.customer.internal_id} is not whitelisted for upload to Loqusdb. Canceling upload for "
+            f"case {case.internal_id}."
+        )
+        raise click.Abort()
+    elif not LinkHelper.all_samples_are_non_tumour(case.links):
```

Comment:
```suggestion
    elif not LinkHelper.is_all_samples_non_tumour(case.links):
```
function for bool return is good practice to start with `is_`